packages/runtime-html/src/resources/custom-attributes/focus.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  bindable,
+  BindingMode,
+  customAttribute,
+  IController,
+  IDOM,
+  INode,
+  State
+} from '@aurelia/runtime';
+import { HTMLDOM } from '../../dom';
+import { addListener, removeListener } from './utils';
+
+/**
+ * Focus attribute for element focus binding
+ */
+@customAttribute('focus')
+export class Focus {
+
+  public static readonly inject: unknown = [INode, IDOM];
+
+  @bindable({ mode: BindingMode.twoWay })
+  public value: unknown;
+
+  private element: HTMLElement;
+  private dom: HTMLDOM;
+
+  /**
+   * Indicates whether `apply` should be called when `attached` callback is invoked
+   */
+  private needsApply: boolean;
+
+  // This is set by the controller after this instance is constructed
+  // tslint:disable-next-line: prefer-readonly
+  private $controller!: IController;
+
+  constructor(
+    element: HTMLElement,
+    dom: HTMLDOM
+  ) {
+    this.element = element;
+    this.dom = dom;
+    this.needsApply = false;
+  }
+
+  public binding(): void {
+    this.valueChanged();
+  }
+
+  /**
+   * Invoked everytime the bound value changes.
+   * @param newValue The new value.
+   */
+  public valueChanged(): void {
+    // In theory, we could/should react immediately
+    // but focus state of an element cannot be achieved
+    // while it's disconnected from the document
+    // thus, there neesd to be a check if it's currently connected or not
+    // before applying the value to the element
+    if (this.$controller.flags & State.isAttached) {
```

Comment:
```suggestion
    if (this.$controller.state & State.isAttached) {
```

`.flags` is `persistentFlags`, has nothing to do with state. Should probably in fact be hidden from public api